config/v1/types_console.go

```diff
@@ -26,6 +28,31 @@ type Console struct {
 type ConsoleSpec struct {
 	// +optional
 	Authentication ConsoleAuthentication `json:"authentication"`
+	// route contains hostname and serving certificate for replacing the default route for the console.
+	// If a custom route is specified, a new route will be created with the provided hostname.
+	// The default console route will be maintained to reserve the default hostname for console if the custom route is removed.
+	// If not specified, default route will be 	used.
+	// +optional
+	Route ConsoleConfigRoute `json:"route,omitempty"`
+}
+
+// ConsoleConfigRoute holds information on external route access to console.
+type ConsoleConfigRoute struct {
+	// name of the route to be created.
+	Name string `json:"name"`
+	// hostname is the desired custom address under which console will be available.
+	// +optional
+	Hostname string `json:"hostname,omitempty"`
```

Comment:
given a different hostname, couldn't they simply point to your service today and configure the route as they wish.  This could make your API contract on the service name, but nothing else.  No new API and no conflict with resources you manage.  What is the value of you having them program this API to program a route instead of simply creating the route?

Reply:
We need to update our OAuth client with the console callback URL. Also the OAuth server whitelists console URL for kubeadmin logout.

We also use cookies and `localStorage`. Multiple routes for console could be problematic since the cookies won't be shared.

Reply:
Console also needs to know its own public URL for various reasons (for instance, to build the OAuth callback URL). Maybe we can read the host from the HTTP request, but I'm not sure that's always reliable.